akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -335,12 +362,21 @@ final class HttpRequest(
 
   /* Manual Case Class things, to easen bin-compat */
 
+  @deprecated("use the method that includes an attributes parameter instead", "10.2.0")
+  private[model] def copy(
+    method:   HttpMethod,
+    uri:      Uri,
+    headers:  immutable.Seq[HttpHeader],
+    entity:   RequestEntity,
+    protocol: HttpProtocol) = new HttpRequest(method, uri, headers, attributes, entity, protocol)
+
   def copy(
```

Comment:
We should make copy private and people should use the `withXYZ` methods which are better to read (imo) and easier to stay compatible with.